src/client/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.1)
 project(TCPServerClient LANGUAGES C)
 
-add_executable(client client.c utils/extract_ip_port_from_string.c utils/opt_arg.c ../lib/check_port.h ../lib/check_port.c ../lib/set_sockaddr_in.h ../lib/set_sockaddr_in.c ../lib/set_sockaddr_in.c)
+add_executable(client client.c utils/extract_ip_port_from_string.c utils/opt_arg.c  ../lib/check_port.c  ../lib/set_sockaddr_in.c)
 
 target_include_directories(client PUBLIC ./)
```

Comment:
In a library directory we have libraries so, why don't you create a directory with check_callbacks and place here check_callbacks.h and build it as a library?

Reply:
Also, you didn't fix last problem, you've included ./ but not ../lib it's unnecessary if you'll make a library, you can just target_link_libraries(client STATIC CheckCallbacks)

Reply:
It's better to explicitly set utils folder and not forcing CMake looking for it!
target_include_directories(client PUBLIC ./ ./utils)